### PR TITLE
Remove hacks that make macOs, Linux, and Windows not fail.

### PR DESCRIPTION
**Pull request by Hixie**
They should fail, because we don't have real support yet.

Also, make the debug override work in release builds, so that people
on those platforms have a workaround.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @stuartmorgan @goderbauer

**Review by goderbauer — APPROVED** [review]
LGTM

Too bad this isn't testable :(

File: packages/flutter/lib/src/foundation/platform.dart [hunk 1]
```diff
@@ -55,12 +72,18 @@ TargetPlatform get defaultTargetPlatform {
...
+
 /// Override the [defaultTargetPlatform].
 ///
 /// Setting this to null returns the [defaultTargetPlatform] to its original
 /// value (based on the actual current platform).
 ///
-/// This setter is only available intended for debugging purposes. To change the
+/// Generally speaking this override is only useful for tests. To change the
 /// target platform in release builds, use the platform override APIs (such as
 /// [ThemeData.platform] in the material library).
```
Comment (stuartmorgan) [inline packages/flutter/lib/src/foundation/platform.dart]: Perhaps this sentence should be tweaked as well? Setting ThemeData.platform doesn't **actually** change the target platform, per discussion, just whether visual theming uses it.

**Review by stuartmorgan-g — COMMENTED** [review]
Thanks, I'll give this a try tonight or tomorrow morning just to make sure it covers the desktop use case!

**Comment by goderbauer** [comment]
@Hixie Should this be reverted and migrated to the new mechanism as well: https://github.com/flutter/flutter/pull/18907? It was added to work around the crash when running on Windows.

**Comment by Hixie** [comment]
if flutter_tester is an internal test feature, running it with FLUTTER_TEST seems fine to me.

**Comment by stuartmorgan** [comment]
Verified that things look good on desktop on the branch after adding the default override.

**Comment by Hixie** [comment]
Will land on green.

**Pull request merged**